jax/interpreters/partial_eval.py

```diff
@@ -1110,8 +1110,9 @@ def process_map(self, map_primitive, f, tracers, params):
     constvars = map(self.getvar, map(self.instantiate_const, consts))
     outvars = map(self.makevar, out_tracers)
     new_in_axes = (None,) * len(consts) + params['in_axes']
-    new_params = dict(params, in_axes=new_in_axes, out_axes=out_axes,
-                      call_jaxpr=convert_constvars_jaxpr(jaxpr))
+    with core.extend_axis_env(axis_name, axis_size, None):  # type: ignore
```

Comment:
Should we put a TODO to revise this? Note from James: collectives that can introduce an axis name into the output, could carry the size in their params.